programs/drift_vaults/src/instructions/initialize_vault.rs

```diff
@@ -25,29 +27,67 @@ pub fn initialize_vault<'info>(
     vault.spot_market_index = params.spot_market_index;
     vault.init_ts = Clock::get()?.unix_timestamp;
 
+    let mut vp_loader = init_vault_protocol(&ctx)?;
+    let vp = vp_loader.as_mut().map(|vp| vp.load_init()).transpose()?;
+
     validate!(
         params.redeem_period < ONE_DAY * 90,
         ErrorCode::InvalidVaultInitialization,
-        "redeem period too long"
+        "redeem period must be < 90 days"
     )?;
     vault.redeem_period = params.redeem_period;
 
     vault.max_tokens = params.max_tokens;
     vault.min_deposit_amount = params.min_deposit_amount;
 
-    validate!(
-        params.management_fee < PERCENTAGE_PRECISION_U64.cast()?,
-        ErrorCode::InvalidVaultInitialization,
-        "management fee must be < 100%"
-    )?;
-    vault.management_fee = params.management_fee;
+    if let (Some(mut vp), Some(vp_params)) = (vp, params.vault_protocol) {
```

Comment:
i think you want to throw error if only one of them is Some

Reply:
Yes